### PR TITLE
Update to TypeScript 3.7

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Mitigates issues in https://github.com/pulumi/pulumi/issues/3606

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@pgavlin @lukehoban @mikhailshilkov any concerns here?

**Comment by lukehoban** [comment]
There is a chance of this breaking users - as it will run a new TS version against all user code.  We definitely want a CHANGELOG entry.  Also - does `^3.7.3` mean we'll float through what TS considers "breaking" versions?  Do we need `~3.7.3` to make sure we float patch version but not minor versions?

Would also love to closely watch `examples` builds after this lands to verify there aren't any failures triggered by this.

File: sdk/nodejs/package.json [hunk 1]
```diff
@@ -20,8 +20,8 @@
         "require-from-string": "^2.0.1",
         "semver": "^6.1.0",
         "source-map-support": "^0.4.16",
-        "ts-node": "^7.0.0",
-        "typescript": "=3.6.3",
+        "ts-node": "^8.5.4",
+        "typescript": "^3.7.3",
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/package.json]:
```suggestion
        "typescript": "~3.7.3",
```

**Comment by CyrusNajmabadi** [comment]
Updated to `~`.  Will add CL entry.  Will watch examples repo once htis goes in.

**Pull request merged**